### PR TITLE
Add CMake fetchcontent documentation and tests

### PR DESCRIPTION
**Pull request by ArthurSonzogni**
Github issue:
https://github.com/nlohmann/json/issues/2073

nlohmann::json documents 2 way of depending on it using CMake
1) Copy-paste the project/source into your own project.
2) Install nlohman::json and then use find_package.

(1) pollutes your git repository, (2) requires everyone to install the
dependencies themselves.

Since 2018, CMake provide some kind of 'package manager' features using
[FetchContent](https://cmake.org/cmake/help/v3.17/module/FetchContent.html)
It gives the following:
~~~cmake
include(FetchContent)

FetchContent_Declare(json
  GIT_REPOSITORY https://github.com/nlohmann/json
  GIT_TAG v3.7.3)

FetchContent_GetProperties(json)
if(NOT json_POPULATED)
  FetchContent_Populate(json)
  add_subdirectory( ${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
endif()
~~~

Then declares the dependency in the target using it:
~~~cmake
target_link_library(my_project PRIVATE nlohmann_json::nlohmann_json
~~~

This patch updates the documentation and provides tests.

### CODE REVIEW

**Comment by ArthurSonzogni** [comment]
Hi Niels,
Could you please take a look?

I was very interested seeing what kind of CI you had. Nice! Let's see how it goes.

Please do not accept this pull request yet. I just realized by cloning this repository that its download size is huge: 170MiB. CMake FetchContent use shallow clone, so this will be only 113MiB.

I think this might be a problem for some users. What do you think about proposing the users to use some alternatives repository URL(s). There are some nice fork tracking only the required files likes:
https://github.com/azadkuh/nlohmann_json_release
Nevertheless, I will have to request them adding a CMakeList.txt to their repository for this to work as a drop-in replacement.

**Comment by nlohmann** [comment]
Thanks! I am already working on splitting tests and code (see https://github.com/nlohmann/json/tree/tests_external), but this will take some more time. Until then, I think we should live with a shallow checkout. It's not perfect, but it works and this is about documenting possibilities for integration - not forcing anyone to actually use this way.

**Comment by ArthurSonzogni** [comment]
The branch "tests_external" looks promising indeed!

----

~~~bash
git clone --depth 1  --branch tests_external https://github.com/nlohmann/json
~~~
=>  4.61 MiB (download size)

----

~~~bash
git clone --depth 1  --branch masterl https://github.com/nlohmann/json
~~~
=> 113 MiB (download size)

----

~~~bash
git clone  --branch masterl https://github.com/nlohmann/json
~~~
=> 170 MiB (download size)

----

File: README.md [hunk 1]
```diff
@@ -133,6 +133,33 @@ add_library(foo ...)
 target_link_libraries(foo PRIVATE nlohmann_json::nlohmann_json)
 ```
 
+##### Embedded (FetchContent)
+Since CMake v3.11,
+[FetchContent](https://cmake.org/cmake/help/v3.11/module/FetchContent.html) can
+be used to automatically download the repository as a dependency.
+
+Example:
+~~~cmake
```
Comment (nlohmann) [inline README.md:142]:
```suggestion
```cmake
```

File: README.md [hunk 2]
```diff
@@ -133,6 +133,33 @@ add_library(foo ...)
...
+  GIT_TAG v3.7.3)
+
+FetchContent_GetProperties(json)
+if(NOT json_POPULATED)
+  FetchContent_Populate(json)
+  add_subdirectory(${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
+endif()
+
+target_link_libraries(foo PRIVATE nlohmann_json::nlohmann_json)
+~~~
```
Comment (nlohmann) [inline README.md:156]:
```suggestion
```
```

File: README.md [hunk 3]
```diff
@@ -133,6 +133,33 @@ add_library(foo ...)
 target_link_libraries(foo PRIVATE nlohmann_json::nlohmann_json)
 ```
 
+##### Embedded (FetchContent)
```
Comment (nlohmann) [inline README.md:136]:
```suggestion
##### Embedded (FetchContent)

```

File: README.md [hunk 4]
```diff
@@ -133,6 +133,33 @@ add_library(foo ...)
...
+Since CMake v3.11,
+[FetchContent](https://cmake.org/cmake/help/v3.11/module/FetchContent.html) can
+be used to automatically download the repository as a dependency.
```
Comment (nlohmann) [inline README.md:139]:
```suggestion
be used to automatically download the repository as a dependency at configure type.
```

File: README.md [hunk 5]
```diff
@@ -133,6 +133,33 @@ add_library(foo ...)
...
+  FetchContent_Populate(json)
+  add_subdirectory(${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
+endif()
+
+target_link_libraries(foo PRIVATE nlohmann_json::nlohmann_json)
+~~~
+
+**Note**: The repository https://github.com/nlohmann/json download size is huge.
+It contains all the dataset used for the benchmarks. You might want to depend on
+a smaller repository. For instance, you might want to replace the URL above by:
```
Comment (nlohmann) [inline README.md:160]:
```suggestion
a smaller repository. For instance, you might want to replace the URL above by
```

**Comment by nlohmann** [comment]
Thanks! I may remove the link to your repo once we have slimmed this repository a bit, see #96.

**Comment by nlohmann** [comment]
* * *

### 🔖 Release item

This issue/PR will be part of the next release of the library. This template helps preparing the release notes.

#### Type

- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] ⚡️ Improvement
- [ ] 🔨 Further Change
- [ ] 🔥 Deprecated function

#### Description

- Added documentation on how import the library via `FetContent`. #2073 #2074

* * *

**Comment by ArthurSonzogni** [comment]
> Thanks! I may remove the link to your repo once we have slimmed this repository a bit, see #96.

Yes sure. This make sense!
Thanks for the review!

**Pull request merged**